### ISSUE TITLE
String_var documentation inconsistencies

### ISSUE THREAD

**Issue by claustromaniac** [llde/Oblivion-Script-Extender#51]
Continuing on with my minor-flaw-nobody-cares-about reporting spree, and in order to not let go of my newly found pleasure of criticising the utterly fantastic work of people more capable than myself, I hereby quote: 

> An uninitialized string variable has a value of zero, which can be tested for in scripts. The value of an initialized string, on the other hand, is undefined and should never be modified directly by statements such as set someStringVar to 6 or using arithmetic operators. Similarly, string variables should only be used to store strings, and **the result of a string-returning variable should only (and always) be assigned to a string variable**.

Array_var implementation followed string_var, so I conclude this is just another sign of outdated documentation. Plus, the wording seems redundant, and maybe even wrong. I guess it would've made more sense in the first place if it said something like...

> the result of a string-returning **function** should only (and always) be assigned to a string variable

On a side note, and hoping you guys are not tired of me already, I was wondering if it would be better if I kept these reports in a single place. I wouldn't want to clutter this space too much, specially considering I'm an uninvited guest. Although I realise it might be a little late for that.

**Comment by Forlini91** [comment]
Even after you fix the statement, there's still a little error: we can have a String returning function and pass it's result directly to a String "eater" command like MessageBox.
I mean:

scn GetAString
Begin Function {}
    ;... do something useful, and finally...
    SetFunctionValue "Hello world!"
    ;A quoted string is returned...
End

//Somewhere else... (compiler override required).
MessageBox $(Call GetAString)

I can guarantee it works as expected, even if it's not clear if OBSE correctly dispose the String.

**Comment by claustromaniac** [comment]
You're right. In fact, you don't even need compiler override for that. I know because I use a function like that to format my debug messages regularly.

However, that newer behaviour _is_ described some paragraphs below. The text contradicts itself there, and then again when it explains how to use array_var (because the result of string-returning functions can be used both as array elements and StringMap keys). Plus, it clearly states in which version was each new feature implemented, which makes me wonder if it is intentionally intended to work as a sort of changelog as well...

Otherwise it should say something like...

> The value of an initialized string, on the other hand, is undefined and should never be modified directly by statements such as set someStringVar to 6 or using arithmetic operators. String variables should only be used to store strings, and the result of a string-returning function should only be assigned to a string variable or array element. < Insert here explanation about using $ to pass functions as arguments >

**Comment by Forlini91** [comment]
I know the compiler override should stay enabled: in my mod VR a bunch of MessageBoxEx exactly like that were showing an empty message with just the dollar sign (literally, the message was saying "$"), and I solved the issue by enabling the compiler override. Without it, the MessageBox $(Call someFn) doesn't work one me and all the users which play the mod.

**Comment by claustromaniac** [comment]
Now that you mention it, I had the same happen to me very recently, but it seems I fixed it by accident when I decided to make a UDF to call MessageEx and MessageBoxEx with SetMessageSound and other things. I did something similar with other functions to be able to pass them OBSE expressions so I wouldn't need to use the compiler override.

What I mean is that you can set up an UDF like this:

```
scn MyMessageBoxUDF
string_var svMsg
Begin Function { svMsg }
	MessageBoxEX "%z" svMsg
End
```

And then you can pass whatever you want to it, like this:

`Call MyMessageBoxUDF ("You died. Why? Because " + (sv_ToUpper $rActor) + " wanted you to die.")`
